dbtc/client/cloud/models/job.py

```diff
@@ -0,0 +1,63 @@
+# stdlib
+from typing import List, Literal, Optional
+
+# third party
+from pydantic import BaseModel
+
+from .constants import State
+
+
+class _JobExecution(BaseModel):
+    timeout_seconds: int
+
+
+class _JobSchedule(BaseModel):
+    cron: str
+    date: Literal['custom_cron', 'days_of_week', 'every_day']
+    time: Literal['every_hour', 'at_exact_hours']
+
+
+class _JobSettings(BaseModel):
+    threads: int
+    target_name: str
+
+
+class _JobTrigger(BaseModel):
+    github_webhook: bool
+    schedule: bool
+    git_provider_webhook: Optional[bool] = None
+
+
+class Job(BaseModel):
```

Comment:
@matt-winkler Here's what I was thinking as far as doing validation for a particular resource.  This is pydantic and it provides this validation out of the box when you provide the schema for the resource.